src/Microsoft.OData.Core/ContainerBuilderExtensions.cs

```diff
@@ -136,7 +136,7 @@ public static IContainerBuilder AddDefaultODataServices(this IContainerBuilder b
             builder.AddService<IJsonWriterFactoryAsync>(ServiceLifetime.Singleton, sp => new DefaultJsonWriterFactory());
             builder.AddService(ServiceLifetime.Singleton, sp => ODataMediaTypeResolver.GetMediaTypeResolver(null));
             builder.AddService<ODataMessageInfo>(ServiceLifetime.Scoped);
-            builder.AddServicePrototype(new ODataMessageReaderSettings());
+            builder.AddServicePrototype(new ODataMessageReaderSettings(odataVersion));
             builder.AddService(ServiceLifetime.Scoped, sp => sp.GetServicePrototype<ODataMessageReaderSettings>().Clone());
             builder.AddServicePrototype(new ODataMessageWriterSettings());

```

Comment:
should it be passed to the writer settings too?

Reply:
ODataMessageWriterSettings doesn't have a constructor to accept ODataVersion.

Reply:
How will the version be passed to the writer settings?

Reply:
or why does the reader settings have a constructor that accepts an ODataVersion but the writer settings doesn't.

Reply:
I have the same concern. Maybe @mikepizzo  has some input.